internal/command/arguments/extended.go

```diff
@@ -128,25 +130,53 @@ func parseTargetables(rawTargetables []string, flag string) ([]addrs.Targetable,
 	return targetables, diags
 }
 
-func parseRawTargetsAndExcludes(targets []string, excludes []string) ([]addrs.Targetable, []addrs.Targetable, tfdiags.Diagnostics) {
+func parseFileTargetables(filePath, flag string) ([]addrs.Targetable, tfdiags.Diagnostics) {
+	var targetables []addrs.Targetable
+	var diags tfdiags.Diagnostics
+
+	// TODO: The actual task of parsing the damn file.
+	// 1. find the damn file and make all the errors for
+	//   tracking it down
+	// 2. line by line for loo
+	// 3. read each target out of the file as a string and
+	//   add into a slice of strings
+	//
+	//
+	// 4. call parseDirectTargetables on overall slice
+	var rawTargetables []string
+	targetables, directDiags := parseDirectTargetables(rawTargetables, flag)
+	diags = diags.Append(directDiags)
+
+	return targetables, diags
+}
+
+func parseRawTargetsAndExcludes(targetsDirect, excludesDirect []string, targetFile, excludeFile string) ([]addrs.Targetable, []addrs.Targetable, tfdiags.Diagnostics) {
 	var parsedTargets []addrs.Targetable
 	var parsedExcludes []addrs.Targetable
 	var diags tfdiags.Diagnostics
 
-	if len(targets) > 0 && len(excludes) > 0 {
+	// Cannot exclude and target in same command
+	if (len(targetsDirect) > 0 || targetFile != "") && (len(excludesDirect) > 0 || excludeFile != "") {
 		diags = diags.Append(tfdiags.Sourceless(
 			tfdiags.Error,
 			"Invalid combination of arguments",
-			"-target and -exclude flags cannot be used together. Please remove one of the flags",
+			"Cannot combine both target and exclude flags. Please only target or exclude resources",
 		))
 		return parsedTargets, parsedExcludes, diags
 	}
 
+	// TODO: this is kinda gross, probably some better coding practices could be applied
+	// Get a review. Potentially just merging the whole damn thing into one, where
+	// parseTargetables accepts all 4 (targetsDirect, excludesDirect, targetFile, excludeFile)
```

Comment:
Comment here says it all. The below feels a bit icky to me, so if you have any suggestions, I'd love to hear.

Reply:
I think doing something like what I suggested in response to the other comment would cause this code to have a slightly different shape, so I'll wait to see how you respond to that before I comment on this particular thing.

Reply:
I wasn't able to get it to a place that the shape was entirely different, so I'll resolve this comment now and re-request your review on this piece.